lib/Sema/CSApply.cpp

```diff
@@ -6288,9 +6288,17 @@ ArgumentList *ExprRewriter::coerceCallArguments(
     // `sending` parameter etc.
     applyFlagsToArgument(paramIdx, argExpr);
 
-    // If the types exactly match, this is easy.
+    auto canShortcutConversion = [&](Type argType, Type paramType) {
+      if (shouldInjectWrappedValuePlaceholder ||
+          paramInfo.hasExternalPropertyWrapper(paramIdx))
+        return false;
+
+      return argType->isEqual(paramType);
+    };
+
     auto paramType = param.getOldType();
-    if (argType->isEqual(paramType) && !shouldInjectWrappedValuePlaceholder) {
+
+    if (canShortcutConversion(argType, paramType)) {
```

Comment:
I do wonder if could do away with this short-circuiting altogether, seems like too easy of a footgun and I'm not sure it's buying us much

Reply:
Yeah, that's what I was thinking as well, I just think it might be worth doing that separately...

Reply:
Oh yeah definitely, doesn't have to be this PR